### ISSUE TITLE
Ensure we have tests triggering various error states in payment disclosure RPC

### ISSUE THREAD

**Issue by str4d** [zcash/zcash#2731]
https://github.com/zcash/zcash/pull/2159#pullrequestreview-76602175

This needs to happen before bringing payment disclosure out of experimental features.

**Comment by daira** [comment]
Bumping.

**Comment by daira** [comment]
Won't fix.
